packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -186,7 +199,36 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
     /**
      * {@inheritDoc (ISharedObject:interface).snapshot}
      */
-    public abstract snapshot(): ITree;
+    public summarize(fullTree: boolean = false): IChannelSummarizeResult {
+        // Create a SummarySerializer that will be used to serialize any IFluidHandles in this object. The
+        // IFluidHandles represents route to any referenced Fluid object.
+        // SummarySerializer tracks the routes of all handles that it serializes.
+        const serializer = new SummarySerializer(this.runtime.channelsRoutingContext);
+
+        const snapshot: ITree = this.snapshotCore(serializer);
+        const summaryTree = convertToSummaryTreeWithStats(snapshot, fullTree);
+
+        const node: IGarbageCollectionNode = {
+            path: this.id,
+            routes: serializer.serializedRoutes,
+        };
+
+        // Return the snapshot tree and the list of routes tracked by the SummarySerializer.
+        return {
+            stats: summaryTree.stats,
+            summary: summaryTree.summary,
+            nodes: [node],
+        };
+    }
+
+    /**
+     * {@inheritDoc (ISharedObject:interface).snapshot}
+     */
+    public snapshot(): ITree {
```

Comment:
Is this left for back compat? If so, can you please mark it with //back-compat and remove it from interface?